scrape_apartments.py

```diff
@@ -21,13 +21,19 @@ def get_apartment_data(base_url, page_number):
         apt_link = listing.find('a', class_='property-link')
         apt_url = apt_link['href'] if apt_link else None
         apt_name = apt_link.get('aria-label', None) if apt_link else None
+        apt_phone_link = listing.find('a', class_='js-phone')
+        apt_phone = apt_phone_link.span if apt_phone_link and apt_phone_link.span else None
         apt_phone_number = None
+        apt_phone_number = apt_phone.get_text(strip=True) if apt_phone else None
+
+
 
         if apt_url:
-            data = requests.get(apt_url, headers=headers)
-            detailsoup = BeautifulSoup(data.text, 'lxml')
-            number = detailsoup.select('.phoneNumber')
-            apt_phone_number = number[0].get_text() if number else None
+            if not apt_phone_number:
```

Comment:
you can change this to and && statement, where you can do

if apt_url AND not apt_phone_number

make sense?